### PR TITLE
Bridge code should use big.Int for transfer quantities in wei

### PR DESCRIPTION
**Pull request by BedrockSquirrel**
### Why is this change needed?

The bridge code was using int64's for quantity in bridge deposits/withdrawals, but ERC20s denominated in 10^-18 "wei" units need to support bigger quantities.

### What changes were made as part of this PR:

- Change bridge/withdraw/deposit code to use big.Int instead of uint64 for quantities
- Update the network validation tests with appropriate units

### :rotating_light: Definition of Done :rotating_light:
- [ ] Deployed into dev-testnet 
- [ ] Passes tests on dev-testnet
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/common/rpc/converters.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func ToRollupHeaderMsg(header *common.Header) *generated.HeaderMsg {
 	var headerMsg generated.HeaderMsg
 	withdrawalMsgs := make([]*generated.WithdrawalMsg, 0)
 	for _, withdrawal := range header.Withdrawals {
-		withdrawalMsg := generated.WithdrawalMsg{Amount: withdrawal.Amount, Recipient: withdrawal.Recipient.Bytes(), Contract: withdrawal.Contract.Bytes()}
+		withdrawalMsg := generated.WithdrawalMsg{Amount: withdrawal.Amount.Bytes(), Recipient: withdrawal.Recipient.Bytes(), Contract: withdrawal.Contract.Bytes()}
```
Comment (BedrockSquirrel) [inline go/common/rpc/converters.go:72]: protobuf doesn't support big.Int directly but we can easily read/write big.Ints from a protobuf `bytes` field.

File: go/ethadapter/erc20contractlib/erc20_contract_lib.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func NewERC20ContractLib(mgmtContractAddr *gethcommon.Address, contractAddrs ...
 }
 
 func (c *erc20ContractLibImpl) CreateDepositTx(tx *ethadapter.L1DepositTx, nonce uint64) types.TxData {
-	data, err := c.contractABI.Pack("transfer", &tx.To, big.NewInt(int64(tx.Amount)))
+	data, err := c.contractABI.Pack("transfer", &tx.To, tx.Amount)
```
Comment (otherview) [inline go/ethadapter/erc20contractlib/erc20_contract_lib.go:57]: It's fine to pack big.Int, yes ?

Comment (BedrockSquirrel) [reply go/ethadapter/erc20contractlib/erc20_contract_lib.go:57]: More than fine, it's required haha. See how before the int64 value was getting converted to big.Int for packing

File: integration/datagenerator/common.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ import (
 	"math"
 	"math/big"
 
+	intcommon "github.com/obscuronet/go-obscuro/integration/common"
```
Comment (otherview) [inline integration/datagenerator/common.go:8]: I'd move this to the go/common, but it's more of a preference.

File: integration/simulation/simulation.go [hunk 4]
```diff
@@ -27,10 +27,11 @@ import (
 )
 
 const (
-	initialBalance  = 5000
 	allocObsWallets = 750000000000000 // The amount the faucet allocates to each Obscuro wallet.
 )
 
+var initialBalance = big.NewInt(5000*10 ^ 18) // balance in wei (1 HOC = 10^18 HOC-wei)
```
Comment (otherview) [inline integration/simulation/simulation.go:33]: Should the converter be used here ?

File: integration/simulation/validate_chain.go [hunk 5]
```diff
@@ -353,7 +358,7 @@ func extractWithdrawals(t *testing.T, nodeClient rpc.Client, nodeAddr uint64) (t
 			return
 		}
 		for _, w := range r.Withdrawals {
-			totalSuccessfullyWithdrawn += w.Amount
+			totalSuccessfullyWithdrawn.Add(totalSuccessfullyWithdrawn, w.Amount)
```
Comment (otherview) [inline integration/simulation/validate_chain.go:361]: Noob question, but don't you have to assign it here ?
`totalSuccessfullyWithdrawn = totalSuccessfullyWithdrawn.Add(totalSuccessfullyWithdrawn, w.Amount)` ?

**Review by otherview — COMMENTED** [review]
Looks good, just a few comments before merging I guess

File: integration/simulation/validate_chain.go (hunk 5 above)
Comment (BedrockSquirrel) [reply integration/simulation/validate_chain.go:361]: Apparently not and seems they recommend not to

**Review by otherview — APPROVED** [review]

**Pull request merged**